# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../lua-wsapi/nil_body.patch
# Copyright (C) 2009 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- ./src/wsapi/common.lua.orig	2009-06-02 17:11:08.000000000 +0200
+++ ./src/wsapi/common.lua	2009-06-02 17:18:51.000000000 +0200
@@ -58,6 +58,9 @@
 
 function send_content(out, res_iter, write_method)
    local write = out[write_method or "write"]
+   if res_iter == nil then
+       return
+   end
    local ok, res = xpcall(res_iter, debug.traceback)
    while ok and res do
       write(out, res)
